packages/lisk-p2p/src/peer.ts

```diff
@@ -44,34 +58,144 @@ export interface PeerConnectionState {
 	readonly outbound: ConnectionState;
 }
 
-export class Peer {
-	private readonly _id: string;
-	private readonly _peerConfig: PeerConfig;
+export class Peer extends EventEmitter {
+	private readonly _handleRPC: (packet: unknown) => void;
+	private readonly _handleMessage: (packet: unknown) => void;
 	private readonly _height: number;
+	private readonly _id: string;
 	private _inboundSocket: any;
-	private _outboundSocket: any;
 	private readonly _ipAddress: string;
+	private _nodeInfo: P2PNodeInfo | undefined;
+	private _outboundSocket: any;
+	private readonly _peerInfo: PeerInfo;
 	private readonly _wsPort: number;
-	private _nodeStatus: P2PNodeStatus | undefined;
 
-	public constructor(peerConfig: PeerConfig) {
-		this._peerConfig = peerConfig;
-		this._ipAddress = peerConfig.ipAddress;
-		this._wsPort = peerConfig.wsPort;
+	public constructor(peerInfo: PeerInfo) {
+		super();
+		this._peerInfo = peerInfo;
+		this._ipAddress = peerInfo.ipAddress;
+		this._wsPort = peerInfo.wsPort;
 		this._id = Peer.constructPeerId(this._ipAddress, this._wsPort);
-		this._inboundSocket = peerConfig.inboundSocket;
-		this._height = peerConfig.height ? peerConfig.height : 0;
+		this._inboundSocket = peerInfo.inboundSocket;
+		if (this._inboundSocket) {
+			this._startListeningForRPCs(this._inboundSocket);
+			this._startListeningForMessages(this._inboundSocket);
+		}
+		this._height = peerInfo.height ? peerInfo.height : 0;
+
+		// This needs to be an arrow function so that it can be used as a listener.
+		this._handleRPC = (packet: unknown) => {
+			// TODO later: Switch to LIP protocol format.
+			// TODO ASAP: Move validation/sanitization to a separate file with other validation logic.
+			const request = packet as ProtocolInboundRPCRequest;
```

Comment:
if we use `as` here, there is no point to use `unknown` as input.
maybe add validation like
```
const isInboundRPCRequest(packet: unknown): packet is ProtocolInboundRPCRequest {}
```

Reply:
There will be an additional function call (probably imported from `response_handler_sanitization.ts`) before the `const request = packet as ProtocolInboundRPCRequest;` line which will do sanitisation on the unknown object.